docs/Toy/toy/rewrites/dead_code_elimination.py

```diff
@@ -0,0 +1,25 @@
+from xdsl.ir import OpTrait, Operation
+from xdsl.pattern_rewriter import RewritePattern, PatternRewriter
+
+
+class Pure(OpTrait):
+    """A trait that signals that an operation has no side effects."""
```

Comment:
I don't think this should be in dead_code eliminiation. I would prefer to have in a separate `traits.py` file, where we provide traits for users to use.

Reply:
Bear in mind that this is in the Toy folder for now, and not yet in xDSL. If you think that this would be a useful addition to xDSL in its current state, I'd be happy to move it there and split it into two files

Reply:
Yeah, I think we should move it before merging this.

Reply:
- [ ] move to xdsl